workspaces/linkerd/plugins/linkerd/src/api/client.ts

```diff
@@ -0,0 +1,23 @@
+import { Entity } from '@backstage/catalog-model';
+import { DiscoveryApi, FetchApi } from '@backstage/core-plugin-api';
+import { DeploymentResponse } from './types';
+
+export class L5dClient {
+  private readonly discoveryApi: DiscoveryApi;
+  private readonly fetchApi: FetchApi;
+
+  constructor(options: { discoveryApi: DiscoveryApi; fetchApi: FetchApi }) {
+    this.discoveryApi = options.discoveryApi;
+    this.fetchApi = options.fetchApi;
+  }
+
+  async getStatsForEntity(entity: Entity): Promise<DeploymentResponse> {
+    return await this.fetchApi
+      .fetch(
+        `${await this.discoveryApi.getBaseUrl('linkerd')}/namespace/${
+          entity.metadata.namespace
+        }/deployment/${entity.metadata.name}/stats`,
+      )
+      .then(r => r.json());
```

Comment:
you may want to consider the 

```ts
    const response = await fetch(...)
    if (!response.ok) {
      throw await ResponseError.fromResponse(response);
    }
    return response.json();
```

to ensure that you get it nicely captured in the standard way with potential stack traces etc. Same in the other places that do `.then(r => r.json())`